src/components/PasswordConfirmation.tsx

```diff
@@ -23,36 +27,55 @@ export function PasswordConfirmation(props: Props) {
   const [t] = useTranslation();
   const navigate = useNavigate();
   const location = useLocation();
+  const company = useCurrentCompany();
 
   const [isModalOpen, setIsModalOpen] = useState(props.show ?? false);
   const [currentPassword, setCurrentPassword] = useState('');
 
-  const isPasswordRequired = true;
+  const [lastPasswordEntryTime, setLastPasswordEntryTime] = useAtom(
+    lastPasswordEntryTimeAtom
+  );
+
+  const lastPwdTimeDiff = dayjs().unix() - lastPasswordEntryTime;
+
+  const isPasswordTimeoutExpired =
+    lastPwdTimeDiff > company.default_password_timeout / 1000 &&
+    company.default_password_timeout > 0;
 
   useEffect(() => {
     setIsModalOpen(props.show as boolean);
   }, [props.show]);
 
-  const handleConfirm = (event: FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    props.onClose(false);
-    props.onSave(currentPassword, isPasswordRequired);
-
+  const handleConfirm = (
+    event?: FormEvent<HTMLFormElement>,
+    isFormConfirmation?: boolean
+  ) => {
+    event?.preventDefault();
+    props.onSave(currentPassword, true);
+    isFormConfirmation && setLastPasswordEntryTime(dayjs().unix());
     setCurrentPassword('');
+
+    props.onClose(false);
   };
 
+  useEffect(() => {
+    if (company && isModalOpen && !isPasswordTimeoutExpired) {
+      handleConfirm();
+    }
+  }, [isModalOpen]);
+
   return (
     <Modal
       onClose={() =>
         location.pathname.startsWith('/settings/users')
           ? navigate('/settings/users')
           : props.onClose(false)
       }
-      visible={isModalOpen}
+      visible={isModalOpen && isPasswordTimeoutExpired}
       title={t('confirmation')}
       text={t('please_enter_your_password')}
     >
-      <form onSubmit={handleConfirm}>
+      <form onSubmit={(event) => handleConfirm(event, true)}>
```

Comment:
Why do we need  second param if it's always true?

Reply:
@beganovich Actually, this parameter is not always true, here is the code:

```
useEffect(() => {
    if (company && isModalOpen && !isPasswordTimeoutExpired) {
      handleConfirm();
    }
  }, [isModalOpen]); 
```

When this function is called on a form, submitting this variable will provide us with the information we need to update the atom with the time of the last password entered. Otherwise, we don't need to store it because that's the time the password hasn't expired.